packages/web-components/src/tablist/tablist.ts

```diff
@@ -0,0 +1,454 @@
+        break;
+      case keyArrowDown:
+        if (this.isHorizontal()) {
+          return;
+        }
+        event.preventDefault();
+        this.adjust(1);
+        break;
+      case keyHome:
+        event.preventDefault();
+        this.adjust(-this.activeTabIndex);
+        break;
+      case keyEnd:
+        event.preventDefault();
+        this.adjust(this.tabs.length - this.activeTabIndex - 1);
+        break;
+    }
+  };
+
+  /**
+   * The adjust method for FASTTabs
+   * @public
+   * @remarks
+   * This method allows the active index to be adjusted by numerical increments
+   */
+  public adjust(adjustment: number): void {
+    if (this.disabled) {
+      return;
+    }
+
+    const focusableTabs = this.tabs.filter(t => isFocusableElement(t));
+    const currentActiveTabIndex = focusableTabs.indexOf(this.activetab);
+
+    const nextTabIndex = wrapInBounds(0, focusableTabs.length - 1, currentActiveTabIndex + adjustment);
+
+    // the index of the next focusable tab within the context of all available tabs
+    const nextIndex = this.tabs.indexOf(focusableTabs[nextTabIndex]);
+
+    if (nextIndex > -1) {
+      this.activateTabByIndex(this.tabs, nextIndex);
+    }
+  }
+
+  private activateTabByIndex(group: HTMLElement[], index: number) {
+    const tab: HTMLElement = group[index] as HTMLElement;
+    this.activetab = tab;
+    this.prevActiveTabIndex = this.activeTabIndex;
+    this.activeTabIndex = index;
+    tab.focus();
+    this.setComponent();
+  }
+
+  private focusTab(): void {
+    this.tabs[this.activeTabIndex].focus();
+  }
+
+  /**
+   * @internal
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    this.tabIds = this.getTabIds();
+    this.activeTabIndex = this.getActiveIndex();
+  }
+}
+
+/**
+ * A BaseTablist component with extra logic for handling the styled active tab indicator.
+ * @public
+ */
+export class Tablist extends BaseTablist {
+  /**
+   * activeTabData
+   * The positional coordinates and size dimensions of the active tab. Used for calculating the offset and scale of the tab active indicator.
+   */
+  private activeTabData: TabData = { x: 0, y: 0, height: 0, width: 0 };
+  /**
+   * previousActiveTabData
+   * The positional coordinates and size dimensions of the active tab. Used for calculating the offset and scale of the tab active indicator.
+   */
+  private previousActiveTabData: TabData = { x: 0, y: 0, height: 0, width: 0 };
+  /**
+   * activeTabOffset
+   * Used to position the active indicator for animations of the active indicator on active tab changes.
+   */
+  private activeTabOffset = 0;
+  /**
+   * activeTabScale
+   * Used to scale the tab active indicator up or down as animations of the active indicator occur.
+   */
+  private activeTabScale = 1;
+
+  /**
+   * appearance
+   * There are two modes of appearance: transparent and subtle.
+   */
+  @attr
+  public appearance?: TablistAppearance = TablistAppearance.transparent;
```

Comment:
Do we want to abstract a base class?

Reply:
So that is already abstracted out of the base class. 

```
export class BaseTablist extends FASTElement {
    activeid: string;
    protected activeidChanged(oldValue: string, newValue: string): void;
    activetab: HTMLElement;
    adjust(adjustment: number): void;
    connectedCallback(): void;
    disabled: boolean;
    protected disabledChanged(prev: boolean, next: boolean): void;
    elementInternals: ElementInternals;
    orientation: TablistOrientation;
    protected orientationChanged(prev: TablistOrientation, next: TablistOrientation): void;
    protected setTabs(): void;
    tabs: HTMLElement[];
    protected tabsChanged(): void;
}
```

And then class extension looks like this... (handles the appearance, size, and custom indicator behaviors)

```
export class Tablist extends BaseTablist {
    activeidChanged(oldValue: string, newValue: string): void;
    appearance?: TablistAppearance;
    protected appearanceChanged(prev: TablistAppearance, next: TablistAppearance): void;
    size?: TablistSize;
    protected sizeChanged(prev: TablistSize, next: TablistSize): void;
    tabsChanged(): void;
}
```

Reply:
Also super weird... because I swore I only saw Tablist class.